src/flowcean/transforms/match_sampling_rate.py

```diff
@@ -46,89 +48,167 @@ class MatchSamplingRate(Transform):
-        """Initialize the MatchSamplingRate transform.
+        """Initialize the transform.
 
         Args:
-            reference_timestamps: Timestamps of the reference feature.
-            feature_columns_with_timestamps: Names of the features that are
-                getting interpolated with their respective original timestamp
-                feature names.
+            reference_feature_name: Reference timeseries feature.
+            feature_interpolation_map: Key-value pairs of the timeseries
+                features that are targeted in interpolation columns and the
+                interpolation method to use. At the moment, the interpolation
+                method can only be 'linear'.
         """
-        self.reference_timestamps = reference_timestamps
-        self.feature_columns_with_timestamps = feature_columns_with_timestamps
+        self.reference_feature_name = reference_feature_name
+        self.feature_interpolation_map = feature_interpolation_map
 
     def apply(self, data: pl.LazyFrame) -> pl.LazyFrame:
-        logger.debug("Matching sampling rate of time series.")
-
-        df = data.collect(streaming=True)
-
-        for i in range(len(df)):
-            reference_timestamps = df[self.reference_timestamps][i]
-            reference_timestamps_np = reference_timestamps.to_numpy()
-            if np.issubdtype(reference_timestamps_np.dtype, np.datetime64):
-                reference_timestamps = (
-                    reference_timestamps_np.astype("datetime64[us]").astype(
-                        np.int64,
-                    )
-                    / 1e6
+        """Transform the input DataFrame.
+
+        Args:
+            data: Input DataFrame.
+
+        Returns:
+            Transformed DataFrame.
+
+        """
+        # preserve all constant columns that are not timeseries data
+        transformed_data = pl.DataFrame()
+        for i in range(len(data.collect().rows())):
+            transformed_data_slice = self._transform_row(
+                data.collect().slice(i, 1),
+            )
+            transformed_data = transformed_data.vstack(transformed_data_slice)
+        return transformed_data.lazy()
+
+    def _transform_row(self, data: pl.DataFrame) -> pl.DataFrame:
+        non_timeseries_features = data.select(
+            pl.exclude(
+                *self.feature_interpolation_map.keys(),
+                self.reference_feature_name,
+            ),
+        )
+        debug_msg = (
+            "Interpolating timeseries features: "
+            f"{self.feature_interpolation_map.keys()} using the timestamps of "
+            f"the and reference feature {self.reference_feature_name}"
+        )
+        logger.debug(debug_msg)
+
+        if self.reference_feature_name not in data.columns:
+            msg = f"'{self.reference_feature_name}' not in the DataFrame."
+            raise FeatureNotFoundError(msg)
+
+        features = list(self.feature_interpolation_map.keys())
```

Comment:
```suggestion
        features = self.feature_interpolation_map.keys()
```

This does not have to be a list, does it?

Reply:
changed